test/rules/use_function_declaration_instead_variable.dart

```diff
@@ -0,0 +1,27 @@
+// Copyright (c) 2017, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+// test w/ `pub run test -N use_function_declaration_instead_variable`
+
+void main() {
+  thisIsGood() { // OK
+    print('this is good');
+  }
+
+  var bad1 = () { // LINT
+    print('this is bad');
+  };
+
+  var bad2 = bad1; // LINT
+
+  bad2 = bad1; // LINT
```

Comment:
This shouldn't be a lint because there's nothing else that the user can write. You can't have
```
var x = ...;
int x(String s) { ... }
```
because then you'd have two definitions of the same name (the local variable declaration and the local function declaration). If the value of `x` needs to change at this point, an assignment is the only option.